File: LeetCode/go/LC2239_FindClosestNumberToZero/find_closest_number_to_zero.go
package find_closest_number_to_zero

import "math"

func findClosestNumber(nums []int) int {
	minAbs := int(math.Abs(float64(nums[0])))
	res := nums[0]
	for i := 1; i < len(nums); i++ {
		abs := int(math.Abs(float64(nums[i])))
		if abs < minAbs {
			minAbs = abs
			res = nums[i]
		} else if abs == minAbs {
			res = max(res, nums[i])
		}
	}
	return res
}

File: LeetCode/java/LC2239_Find_Closest_Number_To_Zero/FindClosestNumberToZero.java
package LC2239_FindClosestNumberToZero;

/**
 * <a href="https://leetcode.cn/problems/find-closest-number-to-zero/">Find Closest Number to Zero</a>
 * 数组
 */
public class FindClosestNumberToZero {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.findClosestNumber(new int[]{-4, -2, 1, 4, 8}));
    }
}

class Solution {
    public int findClosestNumber(int[] nums) {
        int minAbs = Math.abs(nums[0]);
        int res = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int abs = Math.abs(nums[i]);
            if (abs < minAbs) {
                minAbs = abs;
                res = nums[i];
            } else if (abs == minAbs) {
                res = Math.max(res, nums[i]);
            }
        }
        return res;
    }
}